### PR TITLE
azure monitoring (non module)

### PR DESCRIPTION
**Pull request by MalloZup**
## TODOS:

- [x] fix sapahana pattern install @arbulu89
- [x] the monitoring instance in azure cannot contact the hana node via network fix this @MalloZup

### CODE REVIEW

**Comment by MalloZup** [comment]
this PR isn't perfect but I prefer to split up the work and doing smaller prs and make it easier to review.
I will add and improve this later. 

I would not do this if it would be a DESIGN change , but this pr isn't changing the design so we can adopt this strategy to make a not big pr and easy review.

If I can describe this PR:

-  it add documentation, 
- it fix some issues
- it add the monitoring instance in experimental way, I will perfection this later.

File: azure/README.md [hunk 1]
```diff
@@ -1,6 +1,74 @@
...
+- [specification](#specification)
+
+# Quickstart
+
+## 1) Install the azure client 
+
+* [azure commandline](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-zypper?view=azure-cli-latest)
+
+
+### 2) Configuration of terraform:
```
Comment (arbulu89) [inline azure/README.md]: I would remove the `:` of the end

File: azure/README.md [hunk 2]
```diff
@@ -1,6 +1,74 @@
...
+
+1) Rename terraform.tfvars `mv terraform.tfvars.example terraform.tfvars`
+
+2) Generate private and public keys for the cluster nodes with:
+
+```
+mkdir ../salt/hana_node/files/sshkeys; ssh-keygen -t rsa -f ../salt/hana_node/files/sshkeys/cluster.id_rsa
+```
+The key files need to be named as you defined it in `terraform.tfvars` file.
```
Comment (arbulu89) [inline azure/README.md]: Maybe better as:
```The key files must be named as you define them in the `terraform.tfvars` file```

File: azure/README.md [hunk 3]
```diff
@@ -83,7 +136,6 @@ In the file [terraform.tfvars.example](terraform.tfvars.example) there are a num
 * **instancetype**: SKU to use for the cluster nodes; basically the "size" (number of vCPUS and memory) of the VM.
 * **ninstances**: number of cluster nodes to deploy. Defaults to 2.
 * **az_region**: Azure region where to deploy the configuration.
-* **init-type**: initilization script parameter that controls what is deployed in the cluster nodes. Valid values are `all` (installs Hana and configures cluster), `skip-hana` (does not install Hana, but configures cluster) and `skip-cluster` (installs hana, but does not configure cluster). Defaults to `all`.
```
Comment (arbulu89) [inline azure/README.md]: Why have you removed that? Is this variable not existing any longer?

File: azure/image.tf [hunk 4]
```diff
@@ -9,31 +9,49 @@ variable "sles4sap_uri" {
...
-variable "sles4sap_public" {
-  type = map(string)
-
-  default = {
-    "publisher" = "SUSE"
-    "offer"     = "SLES-SAP-BYOS"
-    "sku"       = "12-sp4"
-    "version"   = "2019.03.06"
-  }
+variable "hana_public_publisher" {
```
Comment (arbulu89) [inline azure/image.tf]: Honestly, I preferred the `map` kind of data for this variable. But I don't know how it behaves with terraform 12.
Anyway, It's just a comment

File: azure/image.tf [hunk 5]
```diff
@@ -9,31 +9,49 @@ variable "sles4sap_uri" {
...
-    "offer"     = "SLES-SAP-BYOS"
-    "sku"       = "15"
-    "version"   = "2018.08.20"
-  }
+variable "hana_public_version" {
+  type    = string
+  default = "2019.03.06"
+}
+
+variable "iscsi_publisher" {
```
Comment (arbulu89) [inline azure/image.tf]: shouldn't be `iscsi_public_publisher?

File: azure/image.tf [hunk 6]
```diff
@@ -9,31 +9,49 @@ variable "sles4sap_uri" {
   default = ""
```
Comment (arbulu89) [inline azure/image.tf]: I guess we would need to update the `sles4sap_uri` to `hana_image_uri` or similar. What do you think?

File: azure/monitoring.tf [hunk 7]
```diff
@@ -0,0 +1,29 @@
+variable "monitoring_count" {
+  description = "number of hosts like this one"
```
Comment (arbulu89) [inline azure/monitoring.tf]: Could be improve the description?
`Number of hosts with the monitoring stack installed` or similar maybe?

File: azure/terraform.tfvars.example [hunk 8]
```diff
@@ -55,9 +55,6 @@ hana_inst_folder = "/root/sap_inst/"
 # Device used by node where HANA will be installed
 hana_disk_device = "/dev/sdc"
 
-# Variable for init-nodes.sh script
-init-type = "all"
```
Comment (arbulu89) [inline azure/terraform.tfvars.example]: I don't understand why you are removing this variable yet. As far as I know the QA team was using it. Have you talked with them?

File: salt/pre_installation/ha_repos.sls [hunk 9]
```diff
@@ -8,3 +8,13 @@ ha-factory-repo:
     - retry:
```
Comment (arbulu89) [inline salt/pre_installation/ha_repos.sls]: I think we could improve this state file, just changing the `priority` number depending on if `devel_mode` is set or not. I mean, having only one state, exactly as it was before, and depending on the `devel_mode` use one priority or another

File: azure/image.tf (hunk 4 above)
Comment (MalloZup) [reply azure/image.tf]: I prefer also the map, but I think terraform has an issue with this. TF12 :( . I will open an issue as follow-up if we can investiagate on this

Comment (MalloZup) [reply azure/image.tf]: issue opened as follow-up  https://github.com/SUSE/ha-sap-terraform-deployments/issues/157

File: azure/image.tf (hunk 6 above)
Comment (MalloZup) [reply azure/image.tf]: ah good catch yes

File: azure/terraform.tfvars.example (hunk 8 above)
Comment (MalloZup) [reply azure/terraform.tfvars.example]: I will preserve it. 
thx my cleaning syndrmome :)

File: salt/pre_installation/ha_repos.sls (hunk 9 above)
Comment (MalloZup) [reply salt/pre_installation/ha_repos.sls]: good thing yes make sense

File: azure/image.tf (hunk 6 above)
Comment (MalloZup) [reply azure/image.tf]: I think the rename we can do later. I had look it is quite tangled everywhere..
Perhaps is good name the sles4sap_uri for moment. (e.g we might reuse this image for other things)

**Comment by MalloZup** [comment]
@xabi good point I will need to test this (using the sles4sap_uri) etc.

File: azure/image.tf (hunk 4 above)
Comment (arbulu89) [reply azure/image.tf]: thanks!

**Comment by MalloZup** [comment]
I'm testing currenlty with slessap_uri just in case

**Comment by MalloZup** [comment]
thx! I'm openining follow-up issue for azure so we don't forget about things. I can't fix all the stuff

**Comment by MalloZup** [comment]
https://github.com/SUSE/ha-sap-terraform-deployments/issues/159

**Comment by MalloZup** [comment]
I'm merging this. We will need to figure out how to deal with hawk-api-server

**Comment by MalloZup** [comment]
test with custom image worked

**Pull request merged**